src/gam/__init__.py

```diff
@@ -8327,22 +8327,48 @@ def doRemoveUsersAliases(users):
 
 
 def doUpdateAlias():
+    def verify_alias_target_exists():
+      if target_type != 'group':
+          try:
+              gapi.call(cd.users(), 'get',
+                        throw_reasons=[gapi_errors.ErrorReason.USER_NOT_FOUND],
+                        userKey=target_email)
+              return 'user'
+          except gapi_errors.GapiUserNotFoundError:
+              if target_type == 'user':
+                  return None
+      try:
+          gapi.call(cd.groups(), 'get',
+                    throw_reasons=[gapi_errors.ErrorReason.GROUP_NOT_FOUND],
+                    groupKey=target_email)
+          return 'group'
+      except gapi_errors.GapiGroupNotFoundError:
+          return None
+
     cd = buildGAPIObject('directory')
     alias = normalizeEmailAddressOrUID(sys.argv[3], noUid=True, noLower=True)
     target_type = sys.argv[4].lower()
     if target_type not in ['user', 'group', 'target']:
         controlflow.expected_argument_exit(
             'target type', ', '.join(['user', 'group', 'target']), target_type)
     target_email = normalizeEmailAddressOrUID(sys.argv[5])
-    if len(sys.argv) > 6:
-        myarg = sys.argv[6].lower().replace('_', '')
-        if myarg != 'verifynotinvitable':
+    i = 6
+    while i < len(sys.argv):
+        myarg = sys.argv[i].lower().replace('_', '')
+        if myarg == 'verifynotinvitable':
+            if gapi_cloudidentity_userinvitations.is_invitable_user(alias):
+                controlflow.system_error_exit(51, f'Alias not updated, {alias} is an unmanaged account')
+            i += 1
+        elif myarg == 'verifytarget':
```

Comment:
Suppose we just made this the default behavior? How many API calls would that add to the command (worst and best case scenarios)

Reply:
If you specify user|group, there is one extra call.
If you specify target then there is one extra call if the target is a user, two if the target is a group, two if the target doesn't exist